### PR TITLE
SBT-585: Add support for enabling/customizing gzip configuration at vhost level

### PR DESCRIPTION
**Pull request by syphernl**

### CODE REVIEW

File: nginx/templates/_gzip.jinja [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{%- if gzip is defined and gzip is mapping %}
+{% from "nginx/lib.sls" import configurable %}
+  # Start configure gzip for vhost
+  gzip {{ 'on' if gzip.enable|default(true) else 'off' }};
```
Comment (nstapelbroek) [inline nginx/templates/_gzip.jinja:4]: Can we first test this on a new server before highstating it to every vhost? Setting the default op `false` should fix this.

File: nginx/lib.sls [hunk 2]
```diff
@@ -20,3 +20,13 @@
 
     {{ ret|json() }}
 {%- endmacro -%}
+
+{%- macro configurable(options, key, default_value) -%}
```
Comment (nstapelbroek) [inline nginx/lib.sls:24]: What's this?

File: nginx/templates/_ssl.jinja [hunk 3]
```diff
@@ -1,78 +1,78 @@
 {%- if ssl is defined and ssl is mapping and salt['file.file_exists']('/etc/ssl/certs/' ~ ssl.cert) %}
-# Start SSL support for vhost
-# Listen on SSL
+  # Start SSL support for vhost
```
Comment (nstapelbroek) [inline nginx/templates/_ssl.jinja]: You now have a commit message `SBT-585: Add support for enabling/customizing gzip configuration at vhost level` where you change the indentation of a file. Could maybe be cleaner in a separate commit.

**Review by nstapelbroek — CHANGES_REQUESTED** [review]
Love the feature :100:, I still have some questions / changes before merging.

File: nginx/templates/_gzip.jinja (hunk 1 above)
Comment (syphernl) [reply nginx/templates/_gzip.jinja:4]: It only applies this when the `gzip` key is present in the pillar and if it is a mapping. It will not enable gzip for every vhost we already have :)

Comment (nstapelbroek) [reply nginx/templates/_gzip.jinja:4]: ah check, sweet :+1:

File: nginx/lib.sls (hunk 2 above)
Comment (syphernl) [reply nginx/lib.sls:24]: This deals with a Jinja limitation I ran into regarding falling back to defaults.

```
dict.get('key', 'default')
```

causes an exception if `dict` is unset. Which could be, since someone could enable this feature but not configure it; therefor fall back to the defaults.

```
dict.key|default('default')
```

didn't work either because of the same reason.

I even tried

```
dict.get('key', default')|default('default')
```

which would probably work, but results in repeating myself in regards to default values and doesn't work either if the `dict` is undefined.

File: nginx/templates/_gzip.jinja (hunk 1 above)
Comment (syphernl) [reply nginx/templates/_gzip.jinja:4]: So, essentially you _could_ also do
```yaml
vhost.com:
   gzip:
     options:
        comp_level: 4
```

which doesn't require gzip to be explicitly enabled (`enabled: True`).

File: nginx/templates/_ssl.jinja (hunk 3 above)
Comment (syphernl) [reply nginx/templates/_ssl.jinja]: Done => https://github.com/Enrise/nginx-formula/pull/28

**Review by nstapelbroek — APPROVED** [review]

**Comment by nstapelbroek** [comment]
:tada: lets merge
